rig/routing_table/minimise.py

```diff
@@ -0,0 +1,112 @@
+import collections
+from rig.routing_table import MinimisationFailedError
+from rig.routing_table.remove_default_routes import minimise as \
+    remove_default_entries
+from rig.routing_table.ordered_covering import minimise as ordered_covering
+from six import iteritems
+
+
+def minimise(routing_tables, target_lengths,
+             methods=(remove_default_entries, ordered_covering)):
+    """
```

Comment:
Utility function which attempts to minimises routing tables for multiple chips.

For each routing table supplied, this function will attempt to use the minimisation algorithms given (or some sensible default algorithms), trying each sequentially until a target number of routing entries has been reached.